subprojects/core/src/main/java/org/gradle/internal/jvm/UnsupportedJavaRuntimeException.java

```diff
@@ -33,7 +33,7 @@ public UnsupportedJavaRuntimeException(String message) {
 
     public static void javaDeprecationWarning() {
         if (!JavaVersion.current().isJava8Compatible()) {
-            DeprecationLogger.nagUserWith(JAVA7_DEPRECATION_WARNING, JAVA7_DEPRECATION_WARNING_DOC);
+            DeprecationLogger.nagUserWithDeprecatedInvocationFeature(JAVA7_DEPRECATION_WARNING, JAVA7_DEPRECATION_WARNING_DOC);
```

Comment:
The term "invocation" feels ambiguous. It reminds me too much of "method invocation", which is not what it means here. I have a hard time coming up with an alternative though.

Reply:
Another contender that came up was “environment”. I don't think that's better though.

Reply:
Another option might be “global”.

Reply:
Maybe BuildInvocation to make it unambiguous?

Reply:
environment doesn't really fit as we also use it for deprecated command line options. `BuildInvocation` seems like the best fit here.

Reply:
I can live with `BuildInvocation`.

Reply:
renamed the feature type to BUILD_INVOCATION